src/opt-in.ts

```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+import { commands, WebviewView, WebviewViewProvider } from "vscode";
+
+type OptInMessage = {
+  command: "opt-in";
+};
+
+export class CodingAssistantOptInProvider implements WebviewViewProvider {
+  public static readonly viewType = "sourcery.coding_assistant_opt_in";
+
+  public async resolveWebviewView(webviewView: WebviewView) {
+    webviewView.webview.options = {
+      enableScripts: true,
+    };
+
+    webviewView.webview.onDidReceiveMessage(async (_message: OptInMessage) =>
+      commands.executeCommand("sourcery.coding_assistant.opt_in")
+    );
+
+    webviewView.webview.html = `<!DOCTYPE html>
+<html>
+<head>
+  <title>Opt-In Form</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <style>
+    body {
+      font-family: Arial, sans-serif;
+      margin: 0;
+      padding: 20px;
+    }
+
+    h2 {
+      margin-top: 0px;
+      margin-bottom: 20px;
+    }
+
+    p {
+      margin-bottom: 10px;
+    }
+
+    .opt-in-form {
+      max-width: 400px;
+      margin: 0 auto;
+      padding: 20px;
+      border: 1px solid #ccc;
+      border-radius: 5px;
+    }
+
+    .opt-in-button {
+      display: block;
+      width: 100%;
+      margin-top: 15px;
+      padding: 10px;
+      background-color: #007bff;
+      color: #fff;
+      border: none;
+      border-radius: 5px;
+      cursor: pointer;
+    }
+  </style>
```

Comment:
**suggestion:** consider extracting this to a separate CSS file to match our other webviews.